migrate/dual-write-and-backfill/dual-write-from-postgres.md

```diff
@@ -0,0 +1,262 @@
+time-series data into hypertables. This step consists of identifying those
+tables, excluding their data from the database dump, copying the database
+schema and tables, and setting up the time-series tables as hypertables. The
+data is backfilled into these hypertables in a subsequent step.
+
+<SetupSourceTarget />
+
+### 3a. Dump the database roles from the source database
+
+```
+pg_dumpall -d "$SOURCE" \
+  --quote-all-identifiers \
+  --roles-only \
+  --file=roles.sql
+```
+
+Timescale services do not support roles with superuser access. If your SQL
+dump includes roles that have such permissions, you'll need to modify the file
+to be compliant with the security model.
+
+You can use the following `sed` command to remove unsupported statements and
+permissions from your roles.sql file:
+
+```
+sed -i -E \
+-e '/CREATE ROLE "postgres";/d' \
+-e '/ALTER ROLE "postgres"/d' \
+-e 's/(NO)*SUPERUSER//g' \
+-e 's/(NO)*REPLICATION//g' \
+-e 's/(NO)*BYPASSRLS//g' \
+-e 's/GRANTED BY "[^"]*"//g' \
+roles.sql
+```
+
+<Highlight type="note">
+This command works only with the GNU implementation of sed (sometimes referred
+to as gsed). For the BSD implementation (the default on macOS), you need to
+add an extra argument to change the `-i` flag to `-i ''`.
+
+To check the sed version, you can use the command `sed --version`. While the
+GNU version will explicitly identify itself as GNU, the BSD version of sed
+generally doesn't provide a straightforward --version flag and will simply
+output an "illegal option" error.
+</Highlight>
+
+A brief explanation of this script is:
+
+- `CREATE ROLE "postgres"`; and `ALTER ROLE "postgres"`: These statements are
+  removed because they require superuser access, which is not supported
+  by Timescale.
+
+- `(NO)SUPERUSER` | `(NO)REPLICATION` | `(NO)BYPASSRLS`: These are permissions
+  that require superuser access.
+
+- `GRANTED BY role_specification`: The GRANTED BY clause can also have permissions that
+  require superuser access and should therefore be removed. Note: Per the
+  TimescaleDB documentation, the GRANTOR in the GRANTED BY clause must be the
+  current user, and this clause mainly serves the purpose of SQL compatibility.
+  Therefore, it's safe to remove it.
+
+### 3b. Determine which tables to convert to hypertables
+
+[//]: # (TODO: we should add a reference to timescale doctor here when it is ready)
+
+Ideal candidates for hypertables are large tables containing
+[time-series data][time-series-data].
+This is usually data with some form of timestamp value (`TIMESTAMPTZ`,
+`TIMESTAMP`, `BIGINT`, `INT` etc.) as the primary dimension, and some other
+measurement values.
+
+### 3c. Dump all tables from the source database, excluding data from hypertable candidates
+
+```
+pg_dump -d "$SOURCE" \
+  --format=plain \
+  --quote-all-identifiers \
+  --no-tablespaces \
+  --no-owner \
+  --no-privileges \
+  --exclude-table-data=<table name or pattern> \
```

Comment:
I think we need to provide pattern here, since users do not know about _timescaledb_internal schemas.

Reply:
This is from plain postgres